server/mqtt.go

```diff
@@ -4958,9 +4947,75 @@ func isMQTTReservedSubscription(subject string) bool {
 	return false
 }
 
+func sparkbReplaceDeathTimestamp(msg []byte) []byte {
+	const VARINT = 0
+	const TIMESTAMP = 1
+
+	orig := msg
+	buf := bytes.NewBuffer(make([]byte, 0, len(msg)+16)) // 16 bytes should be enough if we need to add a timestamp
+	writeDeathTimestamp := func() {
+		// [tck-id-conformance-mqtt-aware-ndeath-timestamp] A Sparkplug Aware
+		// MQTT Server MAY replace the timestamp of NDEATH messages. If it does,
```

Comment:
Since all this seem optional, have we decided that we need this feature? I see that you have to "borrow" protobuf implementation (to reduce deps).

Reply:
It is optional in the spec, but the official test that is required to get listed requires it.

2/5 we should support it when SparkplugB-aware. Whether or not configurable altogether with a `sparkplugb: true` - a separate decision?

Reply:
Let's keep the config conversation in the "main" PR, add there if we decide to?